bouncer/shared/polkadot_runtime_update.ts

```diff
@@ -0,0 +1,250 @@
+export async function bumpAndBuildPolkadotRuntime(): Promise<[string, number]> {
+  const projectPath = process.cwd();
+  // tmp/ is ignored in the bouncer .gitignore file.
+  const workspacePath = path.join(projectPath, 'tmp/polkadot');
+  const nextSpecVersion = (await getCurrentRuntimeVersion(POLKADOT_ENDPOINT_PORT)).specVersion + 1;
+  console.log('Current polkadot spec_version: ' + nextSpecVersion);
+
+  // No need to compile if the version we need is the pre-compiled version.
+  const preCompiledWasmPath = `${projectPath}/tests/test_data/polkadot_runtime_${PRE_COMPILED_WASM_VERSION}.wasm`;
+  let copyToPreCompileLocation = false;
+  if (nextSpecVersion === PRE_COMPILED_WASM_VERSION) {
+    if (!fs.existsSync(preCompiledWasmPath)) {
+      console.log(
+        `Warning: Precompiled Wasm file not found at "${preCompiledWasmPath}". It will be compiled and copied there. You will need to commit the file to the repo to speed up future runs.`,
+      );
+      copyToPreCompileLocation = true;
+    } else {
+      console.log(`Using pre-compiled wasm file: ${preCompiledWasmPath}`);
+      return [preCompiledWasmPath, nextSpecVersion];
+    }
+  }
+
+  // Get polkadot source using git
+  if (!fs.existsSync(workspacePath)) {
+    console.log('Cloning polkadot repo to: ' + workspacePath);
+    execSync(`git clone https://github.com/chainflip-io/polkadot.git ${workspacePath}`);
+  }
+  const remoteUrl = execSync('git config --get remote.origin.url', { cwd: workspacePath })
+    .toString()
+    .trim();
+  if (remoteUrl !== POLKADOT_REPO_URL) {
+    throw new Error(
+      `Polkadot folder exists at ${workspacePath} but is not the correct git repo: ${remoteUrl}. Please remove the folder and try again.`,
+    );
+  }
+  console.log('Updating polkadot source');
+  execSync(`git pull`, { cwd: workspacePath });
+
+  await bumpSpecVersion(`${workspacePath}/runtime/polkadot/src/lib.rs`, nextSpecVersion);
+
+  // Compile polkadot runtime
+  console.log('Compiling polkadot...');
+  execSync(`cargo build --locked --release --features fast-runtime`, { cwd: workspacePath });
+  console.log('Finished compiling polkadot');
+  const wasmPath = `${workspacePath}/target/release/wbuild/polkadot-runtime/polkadot_runtime.compact.compressed.wasm`;
+  if (!fs.existsSync(wasmPath)) {
+    throw new Error(`Wasm file not found: ${wasmPath}`);
+  }
+
+  // Backup the pre-compiled wasm file so we do not have to build it again on future fresh runs.
+  if (copyToPreCompileLocation) {
+    fs.copyFileSync(wasmPath, preCompiledWasmPath);
+    console.log(`Copied ${wasmPath} to ${preCompiledWasmPath}`);
+  }
+
+  return [wasmPath, nextSpecVersion];
+}
+
+async function randomPolkadotSwap(): Promise<void> {
+  const assets: Asset[] = [Assets.BTC, Assets.ETH, Assets.USDC, Assets.FLIP];
+  const randomAsset = assets[Math.floor(Math.random() * assets.length)];
+
+  let sourceAsset: Asset;
+  let destAsset: Asset;
+
+  if (Math.random() < 0.5) {
```

Comment:
I like this randomisation aspect, I think we should introduce more randomness like this in the bouncer tests